lib/features/profiles/widgets/profile_header.dart

```diff
@@ -1,34 +1,81 @@
 import 'package:flutter/material.dart';
 import 'package:gruene_app/app/theme/theme.dart';
+import 'package:gruene_app/features/profiles/widgets/profile_image_deleter.dart';
+import 'package:gruene_app/features/profiles/widgets/profile_image_uploader.dart';
 import 'package:gruene_app/swagger_generated_code/gruene_api.swagger.dart';
 
-class ProfileHeader extends StatelessWidget {
+class ProfileHeader extends StatefulWidget {
   final Profile profile;
+  final ValueChanged<Profile> onProfileUpdated;
 
-  const ProfileHeader({super.key, required this.profile});
+  const ProfileHeader({super.key, required this.profile, required this.onProfileUpdated});
+
+  @override
+  State<ProfileHeader> createState() => _ProfileHeaderState();
+}
+
+class _ProfileHeaderState extends State<ProfileHeader> {
+  bool _isProcessing = false;
+
+  void _setProcessing(bool isProcessing) {
+    setState(() {
+      _isProcessing = isProcessing;
+    });
+  }
 
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
+
     return Column(
       crossAxisAlignment: CrossAxisAlignment.center,
       children: [
         SizedBox(
+          width: 90,
           height: 90,
-          child: CircleAvatar(
-            radius: 45,
-            backgroundColor: ThemeColors.textDisabled,
-            backgroundImage: profile.image?.thumbnail.url != null ? NetworkImage(profile.image!.thumbnail.url) : null,
-            child: profile.image?.thumbnail.url == null
-                ? Icon(Icons.person, size: 90, color: theme.colorScheme.surface)
-                : null,
+          child: Stack(
+            alignment: Alignment.center,
+            children: [
+              CircleAvatar(
+                radius: 45,
+                backgroundColor: ThemeColors.textDisabled,
+                backgroundImage: widget.profile.image?.thumbnail.url != null
+                    ? NetworkImage(widget.profile.image!.thumbnail.url)
+                    : null,
+                child: widget.profile.image?.thumbnail.url == null
+                    ? Icon(Icons.person, size: 90, color: theme.colorScheme.surface)
+                    : null,
```

Comment:
:wrench: By extracting the property to an own variable you can make this more readable and avoid the non-null assertions:
``` dart
    final imageUrl = widget.profile.image?.thumbnail.url;
```

```suggestion
                backgroundImage: imageUrl != null ? NetworkImage(imageUrl) : null,
                child: imageUrl == null ? Icon(Icons.person, size: 90, color: theme.colorScheme.surface) : null,
```

Reply:
👍